File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.cpp
class Solution {
public:
    int leastFrequent(vector<int>& freq)
    {
        int ans=INT_MAX;
        for(int i=0;i<26;i++)
        {
            if(freq[i]!=0) ans=min(ans,freq[i]);
        }
        return ans;
    }

    int mostFrequent(vector<int>& freq)
    {
        int ans=INT_MIN;
        for(int i=0;i<26;i++)
        {
            ans=max(ans,freq[i]);
        }
        return ans;
    }

    int beautySum(string s) {
        int n=s.length();
        int totalBeauty=0;
        for(int i=0;i<n;i++)
        {
            vector<int> freq(26,0);
            for(int j=i;j<n;j++)
            {
                freq[s[j]-'a']++;
                int beauty=mostFrequent(freq)-leastFrequent(freq);
                totalBeauty+=beauty;
            }
        }
        return totalBeauty;
    }
};

File: 1890-sum-of-beauty-of-all-substrings/sum-of-beauty-of-all-substrings.java
class Solution {
    public int beautySum(String s) {
        int n = s.length();
        int sum = 0;

        for(int i=0;i<n;i++) {
            int[] hash = new int[26];
            for(int j=i;j<n;j++) {
                int index = s.charAt(j) - 'a';
                hash[index]++;
                int maxFreq = hash[index];
                int minFreq = hash[index];

                for(int k=0;k<26;k++) {
                    maxFreq = Math.max(maxFreq,hash[k]);
                    if(hash[k] > 0) minFreq = Math.min(minFreq,hash[k]);
                }
                sum += (maxFreq - minFreq);
            }
        }
        return sum;
    }
}